packages/subgraph/src/oracles-manager.ts

```diff
@@ -183,3 +218,90 @@ export function handleUpgradeTemplate(event: UpgradeTemplateEvent): void {
     }
     newTemplate.save();
 }
+
+export function handleSetFeatureSetOwner(event: SetFeatureSetOwnerEvent): void {
+    const oraclesManager = getOraclesManager();
+    const templateSet = getTemplateSet(
+        bytesToAddress(oraclesManager.id),
+        event.params.templateId,
+    );
+    templateSet.featuresOwner = event.params.owner;
+    templateSet.save();
+}
+
+export function handleEnableFeatureFor(event: EnableFeatureForEvent): void {
+    const feature = getTemplateFeature(
+        event.params.templateId,
+        event.params.featureId,
+    );
+
+    const allowedFeatureAccountId = feature.id.concat(
+        addressToBytes(event.params.account),
+    );
+    let allowed = OracleTemplateFeatureAllowedAccount.load(
+        allowedFeatureAccountId,
+    );
+    if (allowed !== null) {
+        log.warning(
+            "tried to double enable feature with id {} on template with id {} for user {}",
+            [
+                event.params.featureId.toString(),
+                templateId.toString(),
+                event.params.account.toString(),
+            ],
+        );
+        return;
+    }
+
+    allowed = new OracleTemplateFeatureAllowedAccount(allowedFeatureAccountId);
+    allowed.feature = feature.id;
+    allowed.address = addressToBytes(event.params.account);
+    allowed.save();
+}
+
+export function handleDisableFeatureFor(event: DisableFeatureForEvent): void {
+    const feature = getTemplateFeature(
+        event.params.templateId,
+        event.params.featureId,
+    );
+    const allowedFeatureAccountId = feature.id.concat(
+        addressToBytes(event.params.account),
+    );
+    if (
+        OracleTemplateFeatureAllowedAccount.load(allowedFeatureAccountId) ===
+        null
+    ) {
+        log.warning(
+            "tried to disable feature with id {} on template with id {} for user {} that did not have the feature enabled in the first place",
+            [
+                event.params.featureId.toString(),
+                templateId.toString(),
+                event.params.account.toString(),
+            ],
+        );
+        return;
+    }
+
+    store.remove(
+        "OracleTemplateFeatureAllowedAccount",
+        allowedFeatureAccountId.toString(),
+    );
+}
+
+export function handlePauseFeature(event: PauseFeatureEvent): void {
+    const feature = getTemplateFeature(
```

Comment:
Same here and below for the existence check.

Reply:
Ditto.